RadixWallet/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient+CreateEntity.swift

```diff
@@ -556,12 +570,14 @@ extension OnLedgerEntitiesClient {
 		struct ResourceWithRedemptionValue: Hashable, Sendable {
 			let resource: OnLedgerEntity.Resource
 			var redemptionValue: ResourceAmount?
+			var guarantee: Decimal192?
```

Comment:
Could we have guarantee in ResourceAmount 🤔? As it is another representation of an amount like `exact`, `atLeast`...

Reply:
Hmm, shouldn't guarantee always be an exact amount?

Reply:
Well all types in ResourceAmount contain the ExactAmount, like `case between(minimum: ExactResourceAmount, maximum: ExactResourceAmount)`, and the predicted amount is very similar to that and could be:
```swift
case predicted(predicted: ExactAmount, guaranteed: ExactAmount)
```
The combination of:
```swift
var redemptionValue: ResourceAmount?
var guarantee: Decimal192?
```
Sort of implies that there could be `ResourceAmount.between` + guarantee, which is not true.

Reply:
Right, the `case predicted` is the right way to do it